src/components/Todo/Todo.js

```diff
@@ -4,25 +4,23 @@ import Checkbox from 'components/Checkbox/Checkbox'
 import { FaTrashAlt } from 'react-icons/fa'
 import style from './todo.module.scss'
 
-const Todo = ({ todo, index, handleDeleteIconClick }) => {
+const Todo = ({ todo, index, handleDeleteIconClick, onStatusChange, status }) => {
   const [visible, setVisible] = useState(false)
   const handleIconDelete = () => setVisible(!visible)
 
-  const [statusCheckbox, setStatusCheckbox] = useState('pending')
-
-  const handleCheckboxClick = (event) => {
-    const status = event.target.checked ? 'completed' : 'pending'
-    setStatusCheckbox(status)
-  }
+  const handleCheckboxClick = () => onStatusChange(index)
 
   return (
     <ListItem
-      className={`${style[statusCheckbox]}`}
+      className={`${style.listItem} ${style[status]}`}
       onMouseEnter={handleIconDelete}
       onMouseLeave={handleIconDelete}
       key={index}
     >
-      <Checkbox handleCheckboxClick={handleCheckboxClick}></Checkbox>
+      <Checkbox
+        // checked={(status = 'completed' ? true : false)}
+        onChange={handleCheckboxClick}
+      ></Checkbox>
       {todo}
       {visible && (
         <FaTrashAlt style={{ paddingLeft: '10px' }} onClick={() => handleDeleteIconClick(index)} />
```

Comment:
No uses estilos en línea (poné una clase). El `handler` debería estar declarado fuera del componente, y ese llamar a la función que corresponde (facilita la lectura)

```suggestion
        <FaTrashAlt className={style.deleteIcon} onClick={handleDeleteIconClick} />
```

Y más arriba, fuera del `return` tenés:

```javascript
const handleDeleteIconClick = () => onDelete(index)
```